### PR TITLE
Add cilium-agent and cilium-operator-generic

### PR DESCRIPTION
**Pull request by ysksuzuki**
This PR will be merged after [Prepare Cilium setup #2088](https://github.com/cybozu-go/neco-apps/pull/2088).

It includes

- Install cilium-agent, cilium-operator-generic and hubble-relay.
  -  Cilium won't enforce any policies because policyAuditMode is enabled. It just audits network policies.
  -  Cilium won't proxy packets destined to services because kubeProxyReplacement is disabled.
- Add cilium cli and bubble cli to operation cli.

Confirmed that bootstrap-with-neco-branch has passed.
https://app.circleci.com/pipelines/github/cybozu-go/neco-apps/8660/workflows/8cfabcf1-92b0-435b-8878-7e2b20fccdba/jobs/29054

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -50,6 +55,33 @@ update-coil:
 	bin/kustomize build /tmp/work-coil/cke-patch > etc/coil.yaml
 	rm -rf /tmp/work-coil
 
+.PHONY: update-cilium
+update-cilium: helm
+	$(HELM) repo add cilium https://helm.cilium.io/ > /dev/null
```
Comment (ymmt2005) [inline Makefile:60]: Don't we need `helm repo update`?

Comment (ysksuzuki) [reply Makefile:60]: Thank you for your comment! We do need it.

File: Makefile [hunk 2]
```diff
@@ -50,6 +57,42 @@ update-coil:
...
+		--set hubble.relay.enabled=true \
+		--set hubble.tls.auto.method=cronJob \
+		--set prometheus.enabled=true \
+		--set operator.prometheus.enabled=true \
+		--set hubble.metrics.enabled="{dns,drop:destinationContext=pod|dns|ip;sourceContext=pod|dns|ip,tcp,flow:destinationContext=pod|dns|ip;sourceContext=pod|dns|ip,icmp,http}" \
+		--set rollOutCiliumPods=true > cilium/upstream.yaml
+	sed -i -E '/name:.*cilium$$/!b;n;s/newTag:.*$$/newTag: $(CILIUM_TAG)/' cilium/kustomization.yaml
+	sed -i -E '/name:.*cilium-operator-generic$$/!b;n;s/newTag:.*$$/newTag: $(CILIUM_OPERATOR_TAG)/' cilium/kustomization.yaml
+	sed -i -E '/name:.*hubble-relay$$/!b;n;s/newTag:.*$$/newTag: $(HUBBLE_RELAY_TAG)/' cilium/kustomization.yaml
+	sed -i -E '/name:.*cilium-certgen$$/!b;n;s/newTag:.*$$/newTag: $(CILIUM_CERTGEN_TAG)/' cilium/kustomization.yaml
```
Comment (Hsn723) [inline Makefile:84]: QQ: is there any reason you are using `sed` instead of supplying the image/tag as helm values?

File: Makefile [hunk 3]
```diff
@@ -50,6 +57,42 @@ update-coil:
 	bin/kustomize build /tmp/work-coil/cke-patch > etc/coil.yaml
 	rm -rf /tmp/work-coil
 
+.PHONY: update-cilium
+update-cilium: helm
+	$(HELM) repo add cilium https://helm.cilium.io/ > /dev/null
+	$(HELM) repo update
+	$(HELM) template cilium cilium/cilium --version $(shell echo $(CILIUM_TAG) | cut -d \. -f 1,2,3) \
```
Comment (Hsn723) [inline Makefile:64]: There are a lot of values passed to helm. Would instead storing them in a values.yaml file to pass to helm be better?

File: Makefile (hunk 2 above)
Comment (ysksuzuki) [reply Makefile:84]: I didn't know I could pass the image/tag as helm values.  I'll look into that.

Comment (Hsn723) [reply Makefile:84]: Based on https://github.com/cilium/cilium/blob/master/install/kubernetes/cilium/values.yaml
you should be able to set image/tags like so:
```
--set image.repository=quay.io/cybozu/cilium \
--set image.tag=$(CILIUM_TAG) \
--set certgen.image.repository=quay.io/cybozu/cilium-certgen \
--set certgen.image.tag=$(CILIUM_CERTGEN_TAG) \
--set hubble.relay.image.repository=quay.io/cybozu/hubble-relay \
--set hubble.relay.image.tag=$(HUBBLE_RELAY_TAG) \

...and so on
```

File: Makefile [hunk 4]
```diff
@@ -50,6 +57,36 @@ update-coil:
...
+		--namespace=kube-system \
+		--values cilium/values.yaml \
+		--set image.repository=quay.io/cybozu/cilium \
+		--set image.tag=$(CILIUM_TAG) \
+		--set image.useDigest=false \
+		--set operator.image.repository=quay.io/cybozu/cilium-operator \
```
Comment (ysksuzuki) [inline Makefile:70]: [This](https://github.com/cilium/cilium/blob/915a7f5f4727208bc4b0ff79725781a1f4039d13/install/kubernetes/cilium/templates/cilium-operator/_helpers.tpl#L31) adds a suffix to the repo name. So I omitted 'generic` intentionally.

**Review by Hsn723 — APPROVED** [review]

**Review by zoetrope — APPROVED** [review]
LGTM

**Pull request merged**